### DISCUSSION TITLE
Networking support is limited to using TAP host-side

### DISCUSSION THREAD

**Discussion by codesiddhant** [cloud-hypervisor/cloud-hypervisor#3649] [General]
Most likely we will need to support using 'generic' interfaces on the host as backing devices for our guest emulated network interfaces.

This means being able to use physical interfaces (ethX) as well as virtual interfaces

What if I want to connect 2 VMs which are deployed on different Hosts.
How we can use any overlay network like VxLan/OVS or Something like calico/Cilium on cloud-hypervisor for making any orchestration solution.

## Currently supported list:
- [ ] Physical (ethx)
- [x] Tun/Tap
- [ ] IPsec, PPPoE, GRE
- [ ] VxLan 
- [ ] Other CNI (Calico / Cilium)

## Progress Status
 - [ ] Branch created
 - [ ] Code committed to development branch
 - [ ] Testing - QA
 - [ ] Documentation 
 - [ ] Release notes - resolved issues, compatibility, known issues
  - [ ] Code committed to upstream release/hotfix branches - [ ] Documentation committed to upstream release/hotfix branches

**Comment by rbradford** [comment]
I think you can use DPDK with the vhost-user-net support for all the advanced use cases you outline. If you want to want to have the VM exist on the same network as the host then MacVTap is the solution.

**Reply by codesiddhant** [reply]
is it possible to use any CNI plugin with cloud-Hypervisor?

**Reply by rbradford** [reply]
CNI is about container networking. I don't know how it would apply for a VM. Does QEMU support using CNI plugins?

**Reply by codesiddhant** [reply]
in the Kubernetes cluster, kata-container uses cloud-Hypervisor for running container inside lightweight VM?  
could please explain how they provide a networking interface for those VMs.

**Reply by sboeuf** [reply]
Usually the CNI plugin creates a network interface (let's call it `eth0`) inside the container's network namespace, and Kata Containers runs Cloud Hypervisor or QEMU inside the same network namespace. The VM is created with a TAP interface, and all traffic is redirected between `eth0` and the TAP interface relying on Traffic Control (TC) filtering.

To answer your original question, yes Cloud Hypervisor can be used in the context of Kata Containers along with CNI plugins.

**Reply by codesiddhant** [reply]
ok, thanks for the response.
Please elaborate, is there is any tool available for orchestrating VMs using Cloud-hypervisor between multiple hosts. ?
i.e OpenStack, Kubernetes(kubeVirt), OpenNebula etc.

**Reply by sboeuf** [reply]
There's a Cloud Hypervisor driver in `libvirt`, which means that you should be able to use KubeVirt to manage your VMs.
https://github.com/libvirt/libvirt/blob/master/docs/drvch.rst
https://github.com/libvirt/libvirt/tree/master/src/ch

**Reply by flyncos** [reply]
Unfortunately I don't have time to test it myself, but I think the [tc-redirect-tap](/awslabs/tc-redirect-tap) for Firecracker should work.
